hummingbot/strategy/arbitrage/arbitrage.pyx

```diff
@@ -442,28 +451,76 @@ cdef class ArbitrageStrategy(StrategyBase):
+        # market.c_get_fee returns a namedtuple with 2 keys "percent" and "flat_fees"
+        # "percent" is the percent in decimals the exchange charges for the particular trade
+        # "flat_fees" returns list of additional fees ie: [("ETH", 0.01), ("BNB", 2.5)]
+        # typically most exchanges will only have 1 flat fee (ie: gas cost of transaction in ETH)
         for bid_price_adjusted, ask_price_adjusted, bid_price, ask_price, amount in profitable_orders:
-            # accumulated profitability
-            profitability = (total_bid_value_adjusted + bid_price_adjusted * amount) / \
-                            (total_ask_value_adjusted + ask_price_adjusted * amount)
+            buy_fee = buy_market.c_get_fee(
+                buy_market_symbol,
+                OrderType.MARKET,
+                TradeType.BUY,
+                total_previous_step_base_amount + amount,
+                ask_price
+            )
+            sell_fee = sell_market.c_get_fee(
+                sell_market_symbol,
+                OrderType.MARKET,
+                TradeType.SELL,
+                total_previous_step_base_amount + amount,
+                bid_price
+            )
+            # accumulated flat fees of exchange
+            total_buy_flat_fees = 0.0
+            total_sell_flat_fees = 0.0
+            for buy_flat_fee_currency, buy_flat_fee_amount in buy_fee.flat_fees:
+                if buy_flat_fee_currency == buy_market_quote_currency:
+                    total_buy_flat_fees += buy_flat_fee_amount
+                else:
+                    # if the flat fee currency symbol does not match quote symbol, convert to quote symbol
+                    total_buy_flat_fees += self._exchange_rate_conversion.convert_token_value(
+                        amount=buy_flat_fee_amount,
+                        from_currency=buy_flat_fee_currency,
+                        to_currency=buy_market_quote_currency
+                    )
+            for sell_flat_fee_currency, sell_flat_fee_amount in sell_fee.flat_fees:
+                if sell_flat_fee_currency == sell_market_quote_currency:
+                    total_sell_flat_fees += sell_flat_fee_amount
+                else:
+                    total_sell_flat_fees += self._exchange_rate_conversion.convert_token_value(
+                        amount=sell_flat_fee_amount,
+                        from_currency=sell_flat_fee_currency,
+                        to_currency=sell_market_quote_currency
+                    )
+            total_bid_value_adjusted += bid_price_adjusted * amount
+            total_ask_value_adjusted += ask_price_adjusted * amount
+            # accumulated profitability with fees
+            profitability = (total_bid_value_adjusted * (1 - sell_fee.percent) - total_sell_flat_fees) / \
+                            (total_ask_value_adjusted * (1 + buy_fee.percent) + total_buy_flat_fees)
```

Comment:
I think this is slightly off because we adjust order size to be smaller to be account for fees.
That means the cost of an order should still be the same, but the return is less so the return should be subtracted by both sell and buy fees.

profitability = (total_bid_value_adjusted * (1 - sell_fee.percent) - total_sell_flat_fees - total_buy_flat_fees) / \
                            (total_ask_value_adjusted * (1 + buy_fee.percent))

Reply:
lets sync up on this tomorrow